reactor-core/src/test/java/reactor/core/publisher/FluxReplayTest.java

```diff
@@ -59,6 +62,80 @@ public void vtsStop() {
 		VirtualTimeScheduler.reset();
 	}
 
+	static final Logger log = Loggers.getLogger(FluxReplayTest.class);
+
+	@Test
+	void test() {
+		VirtualTimeScheduler.reset();
+		Flux<Integer> flow = getSource2()
+				.doOnSubscribe(__ -> log.info("Loading source..."))
+				.log("cache")
+				.cache(Duration.ofSeconds(1))
+				.log("actual")
+				.doOnSubscribe(__ -> log.info("Pooling cycle starting..."))
+				.flatMap(this::process, 2)
+				.repeat(1000);
+
+		StepVerifier.create(flow)
+		            .expectNextCount(1001 * 5)
+		            .verifyComplete();
+	}
+
+	private Flux<Integer> getSource2() {
+		return Flux.just(1, 2, 3, 4, 5);
+	}
+
+	private Mono<Integer> process(int channel) {
+		return Mono.just(channel)
+		           .doOnNext(rec -> log.info("Processing: {}", rec))
+		           .delayElement(Duration.ofMillis(50))
+				   .log("Processing: " + channel);
+	}
+
+	@Test
+	void test2() throws InterruptedException {
```

Comment:
please give this test a more demonstrative name

Reply:
FIXED